### PR TITLE
Feature/extract finders from patrol package

### PR DESCRIPTION
**Pull request by bartekpacia**
This is still WIP, all done by @zltnDC. Creating draft PR so it doesn't get lost.

This PR is targeted at #999 and #1273.

### CODE REVIEW

**Comment by bartekpacia** [comment]
We need to:
- update CI
- update docs

File: packages/patrol_finders/README.md
Comment (bartekpacia) [inline packages/patrol_finders/README.md:1]: This README must be updated

File: packages/patrol_finders/example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          xmlns:tools="http://schemas.android.com/tools" package="pl.leancode.patrol.example">
+
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.CAMERA"/>
+    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
+    <uses-permission android:name="android.permission.READ_CONTACTS"/>
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
+    <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" tools:ignore="QueryAllPackagesPermission"/>
```
Comment (bartekpacia) [inline packages/patrol_finders/example/android/app/src/main/AndroidManifest.xml:11]: this app shouldn't have any permissions

File: packages/patrol/lib/src/custom_finders/patrol_integration_tester.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  /// Creates a new [PatrolIntegrationTester] which wraps [tester].
+  const PatrolIntegrationTester({
+    required super.tester,
+    required super.config,
+    required this.nativeAutomator,
+  });
+
+  /// Native automator that allows for interaction with OS the app is running
+  /// on.
+  /// TODO null 3.0
```
Comment (bartekpacia) [inline packages/patrol/lib/src/custom_finders/patrol_integration_tester.dart:23]: What does it mean?

Comment (zltnDC) [reply packages/patrol/lib/src/custom_finders/patrol_integration_tester.dart:23]: #1624

**Comment by bartekpacia** [comment]
The `*-publish.yaml` workflows are very similar. I'll make a PR to create single `publish.yaml`.

File: packages/patrol_finders/example/test/loading_screen_test.dart
Comment (bartekpacia) [inline packages/patrol_finders/example/test/loading_screen_test.dart:1]: Shouldn't we delete tests in `packages/patrol/test` now that they're in `packages/patrol_finders/test`?

Comment (zltnDC) [reply packages/patrol_finders/example/test/loading_screen_test.dart:1]: Which tests do you mean?
https://github.com/leancodepl/patrol/tree/feature/extract-finders-from-patrol-package/packages/patrol/test
I think internals_test.dart is related with the patrol package.

Comment (bartekpacia) [reply packages/patrol_finders/example/test/loading_screen_test.dart:1]: Okay, I must've missed something. The tests I meant are already deleted. Sorry.

File: packages/patrol_finders/example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (bartekpacia) [reply packages/patrol_finders/example/android/app/src/main/AndroidManifest.xml]: well maybe except for `INTERNET`, it's needed for DartVM service

Comment (bartekpacia) [reply packages/patrol_finders/example/android/app/src/main/AndroidManifest.xml]: ah, but `INTERNET` is ususally defined in the `profile` and `debug` variants. nevermind

**Comment by bartekpacia** [comment]
Last thing to do is the documentation, but it can be done later.

For now let's focus on shipping it and making sure it works.

**Pull request merged**